### ISSUE TITLE
[Image] Support loading images from bundles other than MainBundle

### ISSUE THREAD

**Issue by arbesfeld** [facebook/react-native#1762]
There is a [new API](https://developer.apple.com/library/prerelease/ios/documentation/UIKit/Reference/UIImage_Class/#//apple_ref/occ/clm/UIImage/imageNamed:inBundle:compatibleWithTraitCollection:) in iOS 8 that lets us load images from bundles other than the main bundle. Developers might want to use such a feature to asynchronously download bundles for smaller app sizes or to update resources.

We currently default to loading images from the main bundle ([see here](https://github.com/facebook/react-native/blob/master/React/Base/RCTConvert.m#L663)). One idea is to  add a new key to the `source` component where a developer can specify an alternate bundle location:
```
<Image source={{ uri: 'foo', bundle:'~/Documents/async-bundle', width: 300, height: 300 }}/>
```

Thoughts @frantic ?

**Comment by brentvatne** [comment]
Closing because this is inactive, although I am curious about this :) Can you post to ProductPains?

**Comment by arbesfeld** [comment]
This is addressed in #4527 :)
